### PR TITLE
[TS-3]:Integrate analytics and crashlytics

### PR DESCRIPTION
**Pull request by humzacheema**
Jira [TS-3](https://shop-go.atlassian.net/browse/TS-3)

Description:
* Integrated Analytics and Crashlytics.
* Crashes and log events will be logged in firebase.

How to test it?
* call FatalError() in viewDidLoad(), once the app crashes pause the app and reopen the app on stimulator.
* Make an IBaction button and call this method in it Analytics.logEvent("button_pressed", parameters: nil).

ScreenShots&Gifs:
Screen Shot 2022-09-20 at 5 37 15 PM

### CODE REVIEW

File: ShopGo/Base.lproj/Main.storyboard [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
         <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
     </dependencies>
     <scenes>
         <!--View Controller-->
         <scene sceneID="tne-QT-ifu">
             <objects>
-                <viewController id="BYZ-38-t0r" customClass="ViewController" customModuleProvider="target" sceneMemberID="viewController">
+                <viewController id="BYZ-38-t0r" customClass="ViewController" customModule="ShopGo" customModuleProvider="target" sceneMemberID="viewController">
                     <view key="view" contentMode="scaleToFill" id="8bC-Xf-vdC">
-                        <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
```
Comment (basitalich) [inline ShopGo/Base.lproj/Main.storyboard:14]: What changes are made on main story board?

File: ShopGo/ViewController.swift [hunk 2]
```diff
@@ -6,14 +6,12 @@
 //
 
 import UIKit
+import FirebaseAnalytics
 
 class ViewController: UIViewController {
-
+    
     override func viewDidLoad() {
-        super.viewDidLoad()
```
Comment (basitalich) [inline ShopGo/ViewController.swift:13]: why you removed super.viewdidload()?

File: ShopGoTests/ShopGoTests.swift [hunk 3]
```diff
@@ -6,7 +6,6 @@
 //
 
 import XCTest
-@testable import ShopGo
```
Comment (basitalich) [inline ShopGoTests/ShopGoTests.swift:9]: why this is removed?

**Review by basitalich — CHANGES_REQUESTED** [review]

File: ShopGo/Base.lproj/Main.storyboard (hunk 1 above)
Comment (humzacheema) [reply ShopGo/Base.lproj/Main.storyboard:14]: rechecked no changes are made which are visual

File: ShopGoTests/ShopGoTests.swift (hunk 3 above)
Comment (humzacheema) [reply ShopGoTests/ShopGoTests.swift:9]: removed it after checking for errors

File: ShopGo/Base.lproj/Main.storyboard (hunk 1 above)
Comment (basitalich) [reply ShopGo/Base.lproj/Main.storyboard:14]: Then why this file is showing changes?

**Review by basitalich — APPROVED** [review]

**Review by basitalich — DISMISSED** [review]

**Review by haroonmaqsoodch — APPROVED** [review]

**Review by haroonmaqsoodch — APPROVED** [review]

**Review by basitalich — APPROVED** [review]

**Pull request merged**